src/EPPlus/FormulaParsing/Excel/Functions/Engineering/ImReal.cs

```diff
@@ -0,0 +1,34 @@
+﻿/*************************************************************************************************
+  Required Notice: Copyright (C) EPPlus Software AB. 
+  This software is licensed under PolyForm Noncommercial License 1.0.0 
+  and may only be used for noncommercial purposes 
+  https://polyformproject.org/licenses/noncommercial/1.0.0/
+
+  A commercial license to use this software can be purchased at https://epplussoftware.com
+ *************************************************************************************************
+  Date               Author                       Change
+ *************************************************************************************************
+  21/06/2023         EPPlus Software AB       Initial release EPPlus 7
+ *************************************************************************************************/
+using OfficeOpenXml.FormulaParsing.FormulaExpressions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace OfficeOpenXml.FormulaParsing.Excel.Functions.Engineering
+{
+    internal class ImReal : ImFunctionBase
+    {
+        public override CompileResult Execute(IList<FunctionArgument> arguments, ParsingContext context)
+        {
+            GetComplexNumbers(arguments[0].Value, out double real, out double imag, out string imaginarySuffix);
+            if (double.IsNaN(real) || double.IsNaN(imag))
+            {
+                return CompileResult.GetErrorResult(eErrorType.Num);
+            }
+            var result = real;
+            return CreateResult(result.ToString(), DataType.String);
```

Comment:
There result should be a DataType.Decimal. Also remove the ToString() on result to return it as a double instead.